news/10222.bugfix.rst

```diff
@@ -0,0 +1 @@
+Fix ``pip install --pre`` for packages with pre-release build dependencies defined both in ``pyproject.toml`` and ``setup.py``.
```

Comment:
> build dependencies defined both in ``pyproject.toml`` and ``setup.py``

What does defining build dependencies in `setup.py` mean?

Reply:
Having `setup_requires=['my_dependency_package]` in `setup.py`..  as legacy from pre `pyproject.toml` area.